### PR TITLE
Add dark theme toggle

### LINKED ISSUE #14: Add Dark Theme toggle
I want to add a toggle to swap the website to a dark theme. There could also be an option to match the system theme too

### PR DESCRIPTION
**Pull request by Mayank17M**
## Changes

Adds a toggle button to change the website to the dark theme. Also gives an Auto option that displays according to the system preference.
You can see the demo [here](https://youtu.be/JESk9SbaFys).

## Please, go through these steps before you submit a PR.

- [x] My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.

### CODE REVIEW

**Comment by Mayank17M** [comment]
@yashika51 @PAOK-2001 Can you review this?

File: assets/js/toggle.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
+const darkModeButton = document.getElementById("dark-theme-toggle");
+const darkModeStatusIcon = document.getElementById("dark-mode-status-icon");
+const darkModeStatusIndicator = document.getElementById("dark-mode-status-indicator");
+const darkModeStatusIconWrapper = document.getElementById("dark-mode-status-icon-wrapper");
+// Mapping:
+// 0 - auto
+// 1 - dark
+// 2 - light
```
Comment (yashika51) [inline assets/js/toggle.js:8]: Remove this :)

File: assets/js/toggle.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      if (element !== darkModeStatusIcon && element !== darkModeStatusIconWrapper && element !== darkModeStatusIndicator) {
+        element.classList.add("transition-effect")
+      }
+    }
+  }
+  return
+}
+
+async function afterModeChange({ isDark, modePreference, isInit }) {
+  // [TODO] Clean up the animation
```
Comment (yashika51) [inline assets/js/toggle.js:109]: Maybe work on this before merging

**Review by yashika51 — CHANGES_REQUESTED** [review]

File: assets/js/toggle.js (hunk 2 above)
Comment (Mayank17M) [reply assets/js/toggle.js:109]: Sorry, I already did this part. Just forgot to remove the comment.

**Pull request merged**